File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let visited = Array.from({ length: n }, () => 0);

    function bfs(start) {
        let q = [];
        q.push(start);
        visited[start] = 1;

        while (q.length > 0) {
            let now = q.shift();
            for (let next = 0; next < n; next++) {
                if (computers[now][next] === 1 && visited[next] === 0) {
                    visited[next] = 1;
                    q.push(next);
                }
            }
        }
    }

    let answer = 0;
    for (let i = 0; i < n; i++) {
        if (visited[i] === 0) {
            bfs(i);
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [0]*n

    def dfs(current):
        visited[current] = 1
        for i in range(n):
            if visited[i] == 0 and computers[current][i] == 1:
                dfs(i)
                
    for i in range(n):
        if visited[i] == 0:
            answer+=1
            dfs(i)
                
    return answer